config_manager.go

```diff
@@ -35,6 +37,16 @@ type ConfigFile struct {
 	ExcludePatterns *RegexCollection `yaml:"exclude_patterns"`
 }
 
+func (cf ConfigFile) printValidationWarnings() {
+	if len(cf.Files) == 0 {
+		log.Warningf("Validation: configuration file contains no files to watch")
+	}
+	// Warn, that port is not set, only if host is specified
+	if cf.Destination.Host != "" && cf.Destination.Port == 0 {
+		log.Warningf("Validation: destination port is not specified in configuration file")
```

Comment:
I think we should leave the default port (and anything else that currently has a default value). A possible improvement would need to be very meaningful to justify breaking installations which haven't set a port.


On Jun 29, 2015, at 05:12, Leon Sodhi <notifications@github.com> wrote:
> 
> In config_manager.go:
> 
> > @@ -35,6 +37,16 @@ type ConfigFile struct {
> >  	ExcludePatterns *RegexCollection `yaml:"exclude_patterns"`
> >  }
> >  
> > +func (cf ConfigFile) printValidationWarnings() {
> > +	if len(cf.Files) == 0 {
> > +		log.Warningf("Validation: configuration file contains no files to watch")
> > +	}
> > +	// Warn, that port is not set, only if host is specified
> > +	if cf.Destination.Host != "" && cf.Destination.Port == 0 {
> > +		log.Warningf("Validation: destination port is not specified in configuration file")
> @troy as we're not defaulting the destination host to logs.papertrailapp.com, should we also remove the default port and fail hard if it's not specified?
> 
> —
> Reply to this email directly or view it on GitHub.
>